examples/multigpu/node_classification_sage.py

```diff
@@ -132,9 +209,52 @@ def train(
     use_uva,
     num_epochs,
 ):
+    """
+    Training function for GraphSAGE model.
+
+    Parameters:
+    -----------
+    proc_id : int
+        Process ID. This will be a unique number for each spawned process.
+
+    nprocs : int
+        Total number of processes involved in the distributed training.
+
+    device : int
+        The device (e.g., 0) on which to perform computations.
+
+    g : dgl.DGLGraph
+        The input graph on which to perform node classification.
+
+    num_classes : int
+        The number of classes for the classification task.
+
+    train_idx : LongTensor
+        The indices of nodes in the training set.
+
+    val_idx : LongTensor
+        The indices of nodes in the validation set.
+
+    model : SAGE
+        The GraphSAGE model to train.
+
+    use_uva : bool
+        If True, uses Unified Virtual Addressing (UVA) for CUDA computation.
+
+    num_epochs : int
+        The number of epochs for which to train the model.
+
+    Returns:
+    --------
+    None
+    """
+    # Define the neighborhood sampler. The arguments [10, 10, 10]
+    # specify that we sample 10 neighbors at each layer for 3 layers.
+    # Prefetching node features and labels for speedup.
     sampler = NeighborSampler(
         [10, 10, 10], prefetch_node_feats=["feat"], prefetch_labels=["label"]
     )
+
     train_dataloader = DataLoader(
```

Comment:
This is a place that might cause confusion for users. The dataloader was created in different processs but the initiation does not take proc_id as parameter. It worth to callout how it was handled.